parity-crypto/src/publickey/secret_key.rs

```diff
@@ -224,18 +252,21 @@ impl FromStr for Secret {
 impl From<[u8; 32]> for Secret {
 	#[inline(always)]
 	fn from(mut k: [u8; 32]) -> Self {
-		let result = Secret { inner: Box::new(H256(k)) };
+		let mut me = Self::zero();
+		let mut secret = H256(k);
+		me.inner.as_bytes_mut().swap_with_slice(secret.as_bytes_mut());
 		k.zeroize();
-		result
+		me
 	}
 }
 
 impl From<H256> for Secret {
 	#[inline(always)]
 	fn from(mut s: H256) -> Self {
-		let result = s.0.into();
+		let mut me = Self::zero();
+		me.inner.as_bytes_mut().swap_with_slice(s.as_bytes_mut());
 		s.0.zeroize();
```

Comment:
zeroize is probably redundant here as we swap bytes with `Self::zero`, but it's not clear to me if LLVM can optimize swapping with zeroes somehow

Reply:
I am not sure why we use swap_with_slice, copy_from_slice looks more straight forward to me and could not have an internal buffer (I did not check for copy but swap uses one).

Reply:
But having H256 on stack, I would also expect it to be zeroize by swap_with_slice.

Reply:
You're indeed right that `swap_with_slice` uses a small internal buffer (u64 * 4): https://doc.rust-lang.org/src/core/ptr/mod.rs.html#447, although LLVM should be able optimize it as you can see from the comments.

I can change it to `copy_from_slice` and `zeroize`

Reply:
I think copy_from_slice then zeroize is clearer about the intent (but swap alone seems fine too, just swap and zeroize at the same time seems odd as you mention).

Reply:
> copy_from_slice then zeroize is clearer about the intent

I agree. 👍

Reply:
fixed in d628053